test/e2e/parallel/spacerequest_test.go

```diff
@@ -87,6 +90,97 @@ func TestCreateSpaceRequest(t *testing.T) {
 				_, err = awaitilities.Host().WaitForSpace(t, subSpace.GetName(),
 					UntilSpaceHasTier("appstudio-env"), // tierName is back as the one on spaceRequest
 					UntilSpaceHasTargetClusterRoles(targetClusterRoles),
+					UntilSpaceHasDisableInheritance(false),
+					UntilSpaceHasAnyProvisionedNamespaces(),
+				)
+				require.NoError(t, err)
+
+				t.Run("delete space request", func(t *testing.T) {
+					// now, delete the SpaceRequest and expect that the Space will be deleted as well,
+					// along with its associated namespace
+
+					// when
+					err := memberAwait.Client.Delete(context.TODO(), spaceRequest)
+
+					// then
+					// subSpace should be deleted as well
+					require.NoError(t, err)
+					err = memberAwait.WaitUntilNamespaceDeleted(t, subSpace.Name, "appstudio-env")
+					require.NoError(t, err)
+					err = memberAwait.WaitUntilNSTemplateSetDeleted(t, subSpace.Name)
+					require.NoError(t, err)
+					err = hostAwait.WaitUntilSpaceAndSpaceBindingsDeleted(t, subSpace.Name)
+					require.NoError(t, err)
+				})
+			})
+		})
+	})
+
+	t.Run("create space request with DisableInheritance", func(t *testing.T) {
+		// when
+		targetClusterRoles := []string{cluster.RoleLabel(cluster.Tenant)}
+		spaceRequest, parentSpace := CreateSpaceRequest(t, awaitilities, memberAwait.ClusterName,
+			WithSpecTierName("appstudio-env"),
+			WithSpecTargetClusterRoles(targetClusterRoles),
+			WithSpecDisableInheritance(true))
+
+		// then
+		// check for the subSpace creation
+		subSpace, err := awaitilities.Host().WaitForSubSpace(t, spaceRequest.Name, spaceRequest.Namespace, parentSpace.GetName(),
+			UntilSpaceHasTargetClusterRoles(targetClusterRoles),
+			UntilSpaceHasTier("appstudio-env"),
+			UntilSpaceHasDisableInheritance(true),
+			UntilSpaceHasAnyProvisionedNamespaces(),
+		)
+		require.NoError(t, err)
+		subSpace, _ = VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name, UntilSpaceHasAnyTargetClusterSet())
+		spaceRequest, err = memberAwait.WaitForSpaceRequest(t, types.NamespacedName{Namespace: spaceRequest.GetNamespace(), Name: spaceRequest.GetName()},
+			UntilSpaceRequestHasConditions(Provisioned()),
+			UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint),
+			UntilSpaceRequestHasNamespaceAccess(subSpace),
+		)
+		require.NoError(t, err)
+		VerifyNamespaceAccessForSpaceRequest(t, memberAwait.Client, spaceRequest)
+
+		t.Run("subSpace is recreated if deleted ", func(t *testing.T) {
```

Comment:
I included them because I wasn't not 100% sure what they do and thought it may be best to check that disable inheritance being true has not affected them. Would be delighted to remove them if you think best :-)

Reply:
I think we can remove those. I don't think the disableInheritance option influences this path ( the fact that the subSpace is recreated with the config from SpaceRequest ).

Reply:
we could update the disableInheritance field in the subSpace as part of "subSpace always reflects values from spaceRequest" and check that the spacerequest controller puts that back to it's original value.